File: persistence/src/main/java/io/sparkled/persistence/sequence/impl/query/DeleteRenderedStagePropsQuery.java
package io.sparkled.persistence.sequence.impl.query;

import io.sparkled.model.entity.RenderedStageProp;
import io.sparkled.persistence.PersistenceQuery;

import javax.persistence.EntityManager;
import javax.persistence.Query;
import java.util.logging.Logger;

public class DeleteRenderedStagePropsQuery implements PersistenceQuery<Void> {

    private static final Logger logger = Logger.getLogger(DeleteRenderedStagePropsQuery.class.getName());

    private final int sequenceId;

    public DeleteRenderedStagePropsQuery(int sequenceId) {
        this.sequenceId = sequenceId;
    }

    @Override
    public Void perform(EntityManager entityManager) {
        String className = RenderedStageProp.class.getSimpleName();
        Query query = entityManager.createQuery("delete from " + className + " where sequenceId = :id");
        query.setParameter("id", sequenceId);

        int deleted = query.executeUpdate();
        logger.info("Deleted " + deleted + " " + className + " record(s) for sequence " + sequenceId);
        return null;
    }
}

File: persistence/src/main/kotlin/io/sparkled/persistence/query/stage/DeleteRenderedStagePropsQuery.kt
package io.sparkled.persistence.query.stage

import com.fasterxml.jackson.databind.ObjectMapper
import io.sparkled.persistence.DbQuery
import org.jdbi.v3.core.Jdbi
import org.slf4j.LoggerFactory

class DeleteRenderedStagePropsQuery(
    private val renderedStagePropIds: Collection<Int>
) : DbQuery<Unit> {

    override fun execute(jdbi: Jdbi, objectMapper: ObjectMapper) {
        jdbi.perform { handle ->
            val rowsDeleted = handle
                .createUpdate(query)
                .bindList("ids", renderedStagePropIds)
                .execute()

            logger.info("Deleted $rowsDeleted rendered stage prop(s).")
        }
    }

    companion object {
        private val logger = LoggerFactory.getLogger(DeleteRenderedStagePropsQuery::class.java)
        private const val query = "DELETE FROM rendered_stage_prop WHERE id IN (<ids>)"
    }
}
